src/vmspawn/vmspawn.c

```diff
@@ -2022,7 +2025,21 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
                 if (!escaped_drive)
                         return log_oom();
 
-                r = strv_extendf(&cmdline, "driver=raw,cache.direct=off,cache.no-flush=on,file.driver=file,file.filename=%s,node-name=vmspawn_extra_%zu", escaped_drive, i);
+                fd = open(*drive, O_RDONLY | O_CLOEXEC);
```

Comment:
any reason you are actually opening this here?

i mean, I am a big fan of pinning inodes via fds to avoid races. But you are not really doing this here: you open the inode, and the only op you do is fstat()... You might as well have used plain stat() instead of fstat() hence?

I mean, if we could pass an fd to the device to qemu (instead of a path), that'd be great, but i think we cannot actually do that. or can we? if we could then it would be great if we open the thing once, then do our check via it, and then pass it to qemu as open fd.